streams/src/test/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImplTest.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.kstream.KGroupedStream;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.KStreamBuilder;
+import org.apache.kafka.streams.kstream.TimeWindows;
+import org.apache.kafka.test.MockAggregator;
+import org.apache.kafka.test.MockInitializer;
+import org.apache.kafka.test.MockReducer;
+import org.junit.Before;
+import org.junit.Test;
+
+public class KGroupedStreamImplTest {
+
+    private KGroupedStream<String, String> groupedStream;
+
+    @Before
+    public void before() {
+        final KStream<String, String> stream = new KStreamBuilder().stream("topic");
+        groupedStream = stream.groupByKey();
+    }
+
+    @Test(expected = NullPointerException.class)
```

Comment:
Yup, I understand that they are checking in fact throw NPE, but pass a null and get a NPE, is this expeced result? Or it just broke down the line). Get it?

If those tests raised a IllegalArgumentException when passing null then now we have something more meaningful. But I think it's completely valid to keep those tests as-is in this PR. :) They are worth keeping mainly because they improve coverage of the code, so I agree that they are valid. Worth having? I dunno in the long term.*